CASTable docstring edits

kybowm opened the pull request:
Here are some changes to CASTable docstrings. There were no changes to functional code.

None of the docstring edits are critical, but since the CASTable API is absorbed by other products like dlpy.ImageTable, I wanted to make them stylistically similar. Style is much less important than function, though. If these suggestions don't effectively communicate how swat works, I'm open to changes.

kesmit13 commented on swat/cas/table.py:
```diff
@@ -2784,7 +2783,7 @@ def head(self, n=5, columns=None, bygroup_as_index=True, casout=None):
             A subset of columns to return.
         bygroup_as_index : boolean
-            If By groups are specified, should they be converted to an index?
+            When By groups are specified, if bygroup_as_index is True, By groups are converted to an index
```
I'm not fond of the wording here.  Maybe "When By groups are specified and bygroup_as_index is True, By groups are converted to an index.  Also, lines should be kept to <78 characters.

kybowm replied:
When by_group_index is True, By groups are converted to an index if they exist

kesmit13 commented on swat/cas/table.py:
```diff
@@ -2813,7 +2812,7 @@ def tail(self, n=5, columns=None, bygroup_as_index=True, casout=None):
             A subset of columns to return.
         bygroup_as_index : boolean
-            If By groups are specified, should they be converted to an index?
+            When By groups are specified, if bygroup_as_index is True, By groups are converted to an index
```
Same as above

kybowm replied:
Same as above

kesmit13 commented on swat/cas/table.py:
```diff
@@ -2846,7 +2845,7 @@ def slice(self, start=0, stop=None, columns=None, bygroup_as_index=True, casout=
             A subset of columns to return.
         bygroup_as_index : boolean
-            If By groups are specified, should they be converted to an index?
+            When By groups are specified, if bygroup_as_index is True, By groups are converted to an index
```
Same as above

kybowm replied:
Same as above

kesmit13 commented on swat/cas/table.py:
```diff
@@ -3425,7 +3424,7 @@ def _materialize(self, casout=None, inplace=False, prefix=None, suffix=None):
             The CAS output table definition
         inplace : bool, optional
-            Should the output table be overwritten?
+            If True, the output table is overwritten
```
You might want to add "if it exists" at the end

kybowm replied:
Changed as suggested.

kesmit13 commented on swat/cas/table.py:
```diff
@@ -3506,7 +3505,7 @@ def _bool(self):
     def all(self, axis=None, bool_only=None, skipna=True, level=None, **kwargs):
         '''
-        Return whether all elements in the column are True
+        Return True for each column with only True elements
```
This wording makes it sound like the items have to literally be True values, but it's really any value that is treated as true.  Maybe just not capitalize the second "True".

kybowm replied:
Return True for each column with only elements treated as True

kesmit13 commented on swat/cas/table.py:
```diff
@@ -3555,7 +3554,7 @@ def all(self, axis=None, bool_only=None, skipna=True, level=None, **kwargs):
     def any(self, axis=None, bool_only=None, skipna=True, level=None, **kwargs):
         '''
-        Return whether any elements in the column are True
+        Return True for each column with at least one True element
```
Don't capitalize second "True".

kybowm replied:
Return True for each column with at least one element treated as True

kesmit13 replied:
I'm still not fond of the second literal 'True' here.  The values don't have to literally be 'True', they just have to be true by rules of truth (i.e., true == non-zero / non-empty-string / etc.).

kybowm replied:
Okay. I changed the second 'True' to 'true' in any() and all().

kesmit13 commented on swat/cas/table.py:
```diff
@@ -7798,20 +7796,20 @@ def capitalize(self):
     def contains(self, pat, case=True, flags=0, na=np.nan, regex=True):
         '''
-        Does the value contain the specified pattern?
+        Indicates wheter the value contains the specified pattern.
```
Mis-spelled "whether"

kybowm replied:
Fixed

kesmit13 commented on swat/cas/table.py:
```diff
@@ -7798,20 +7796,20 @@ def capitalize(self):
     def contains(self, pat, case=True, flags=0, na=np.nan, regex=True):
         '''
-        Does the value contain the specified pattern?
+        Indicates wheter the value contains the specified pattern.
```
No period at end.  This should be done for all of the one-line descriptions of methods.

kybowm replied:
Fixed for all.

kesmit13 commented on swat/cas/table.py:
```diff
@@ -7865,20 +7863,20 @@ def count(self, pat, flags=0, **kwargs):
     def endswith(self, pat, case=True, flags=0, na=np.nan, regex=True):
         '''
-        Does the table column end with the given pattern?
+        Indicates whether the table column ends with the given pattern.
```
No period at end

kybowm replied:
Fixed.

kesmit13 commented on swat/cas/table.py:
```diff
@@ -8176,20 +8174,20 @@ def slice(self, start=0, stop=None, step=None):
     def startswith(self, pat, case=True, flags=0, na=np.nan, regex=True):
         '''
-        Does the table column start with the given pattern?
+        Indicates wheter the table column start with the given pattern.
```
Mis-spelled "whether"

kybowm replied:
Fixed

kesmit13 commented on swat/cas/table.py:
```diff
@@ -8283,7 +8281,7 @@ def isalpha(self):
     def isdigit(self):
         '''
-        Does the value contain only digits?
+        Indicates wheter the value contains only digits.
```
"whether"

kybowm replied:
Fixed

kesmit13 commented on swat/cas/table.py:
```diff
@@ -8328,7 +8326,7 @@ def islower(self):
     def isupper(self):
         '''
-        Does the value contain only uppercase characters?
+        Does the value contains only uppercase characters.
```
Put question mark back in.  The description is phrased as a question.

kybowm commented:
@kesmit13 I forgot I had this PR open. I believe I have addressed all your suggestions. I changed many as you suggested, but for some I proposed alternatives. I also updated my fork with the recent changes to swat. Let me know what you think of the edits.

kesmit13 commented on swat/cas/table.py:
```diff
@@ -126,7 +126,7 @@ def _get_unique(seq, lowercase=False):
     ----------
     lowercase : boolean
-        Should values be compared in a case-insensitive way?
+        If true, compare elements in a case-insensitive way
```
This 'true' should be capitalized to match the other changes.  After this change, I'm fine with the whole set of commits.

kybowm replied:
Sounds good. I just updated it. Thanks!
